reference-infrastructure/storage/nfs/nfs-server.yaml

```diff
@@ -0,0 +1,102 @@
+---
+
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: io1-fast
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "false"
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: io1
+  iopsPerGB: "10"
+  encrypted: "true"
+  fsType: ext4
+  
+---
+
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: nfs-server-data-pvc
+spec:
+  storageClassName: io1-fast
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 100Gi
+
+---
+
+kind: Service
+apiVersion: v1
+metadata:
+  name: nfs-server
+spec:
+  type: LoadBalancer
+  ports:
+    - name: nfs
+      port: 2049
+    - name: mountd
+      port: 20048
+    - name: rpcbind
+      port: 111
+  selector:
+    role: nfs-server
+
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nfs-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      role: nfs-server
+  template:
+    metadata:
+      labels:
+        role: nfs-server
+    spec:
+      containers:
+        - name: nfs-server
+          image: k8s.gcr.io/volume-nfs:0.8
+          imagePullPolicy: Always
```

Comment:
should we change the pull policy to `IfNotPresent`?

Reply:
Also, it looks like that image hasn't been touched for 3 years, which means it will be behind a lot of security updates (and fixes in general). The previous approach (although clearly more complicated) had the advantage of using a fresh, up to date image. I think we should either ship the previous solution, or, if we want something simpler, we should ship a remade image.

Reply:
@pbruski you are correct. The `centos:latest` image contains 4 vulnerabilities (per `snyk scan`). It is doable to update them with
```
RUN yum update -y openssl-libs bind-export-libs gnutls
```
but I am not sure if we want to create a new Docker image. We were pondering this together with @dylanrathbone and we are leaning towards creating our own image based on a similar service approach you did in the `scripts/nfs` while using this new template. I don't think it will be still a production-ready approach but I am not sure if we can advise customers on a production-grade NFS server.

Reply:
That sounds good - yeah, I agree that producing an image as an official artifact would be problematic.